python_modules/libraries/dagster-aws/dagster_aws/secretsmanager/resources.py

```diff
@@ -1,22 +1,62 @@
 from contextlib import contextmanager
+from typing import Any, Dict, Generator, List, Optional
 
 from dagster import (
-    Array,
-    Field,
-    Noneable,
-    _check as check,
     resource,
 )
+from dagster._core.execution.context.init import InitResourceContext
 from dagster._core.test_utils import environ
 from dagster._utils.merger import merge_dicts
+from pydantic import Field
 
-from dagster_aws.utils import BOTO3_SESSION_CONFIG
+from dagster_aws.utils import ResourceWithBoto3Configuration
 
 from .secrets import construct_secretsmanager_client, get_secrets_from_arns, get_tagged_secrets
 
 
-@resource(BOTO3_SESSION_CONFIG)
-def secretsmanager_resource(context):
+class SecretsManagerResource(ResourceWithBoto3Configuration[Any]):
```

Comment:
Given the comment, can we type this as `ResourceWithBoto3Configuration[botocore.client.SecretsManager]`

Reply:
Added these as a string type annotation - the actual types don't load in runtime (so they need to be string annotations), but I believe if a user installs something like https://pypi.org/project/types-botocore/ it will have correct hints.